### PR TITLE
Sudoku solver using Backtracking

### PR DESCRIPTION
**Pull request by rgrupesh**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes.)

### Description

(Implemented Sudoku Solver using backtracking in Python.)

### Checklist

> * [x]  I've read the contribution guidelines.
> * [x] I've checked the issue list before deciding what to submit.
> * [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

For Issue: [#622]

### CODE REVIEW

**Comment by rgrupesh** [comment]
@atarax665

**Comment by vatsalkesarwani12** [comment]
resolve the branch conflict

**Comment by rgrupesh** [comment]
> resolve the branch conflict

resolved.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "/bin/python3"
```
Comment (vatsalkesarwani12) [inline .vscode/settings.json]: delete this file

File: Python/README.md [hunk 2]
```diff
@@ -100,6 +100,260 @@ _add list here_
...
+* [Bubble Sort](sort/Bubble_Sort.py)
+* [Heap Sort](sort/Heap_sort.py)
+* [Insertion Sort](sort/insertion.py)
+* [Merge Sort](sort/Merge_Sort.py)
+* [Quick Sort](sort/QuickSort.py)
+* [Radix Sort](sort/Radix_Sort.py)
+* [Selection Sort](sort/Selection_Sort.py)
+* [Topological Sort](sort/topological_sort.py)
+* [Intro Sort](sort/IntroSort.py)
+
```
Comment (vatsalkesarwani12) [inline Python/README.md]: why so many changes in this file

File: Python/other/sudoku_backtracking.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+board = [[5, 3, 0, 0, 7, 0, 0, 0, 0],
```
Comment (vatsalkesarwani12) [inline Python/other/sudoku_backtracking.py]: take dynamic input with proper message

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (rgrupesh) [reply .vscode/settings.json]: > delete this file

done

File: Python/other/sudoku_backtracking.py (hunk 3 above)
Comment (rgrupesh) [reply Python/other/sudoku_backtracking.py]: > take dynamic input with proper message

done

File: Python/other/sudoku_backtracking.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            if board[new_x + i][new_y + j] == n:
+                return False
+
+    return True 
+
+if __name__ == "__main__":
+    print("Enter the numbers row by row, and put 0 for empty space:")
+    board = [[int(input()) for x in range (9)] for y in range(9)] 
+    solve(board)  
+    print_board(board)
```
Comment (vatsalkesarwani12) [inline Python/other/sudoku_backtracking.py]: add sample test case in comments

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

File: Python/other/sudoku_backtracking.py (hunk 4 above)
Comment (rgrupesh) [reply Python/other/sudoku_backtracking.py]: done

File: Python/other/sudoku_backtracking.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                   [6, 0, 0, 1, 9, 5, 0, 0, 0], 
+                   [0, 9, 8, 0, 0, 0, 0, 6, 0],
+                   [8, 0, 0, 0, 6, 0, 0, 0, 3],
+                   [4, 0, 0, 8, 0, 3, 0, 0, 1],
+                   [7, 0, 0, 0, 2, 0, 0, 0, 6],
+                   [0, 6, 0, 0, 0, 0, 2, 8, 0],
+                   [0, 0, 0, 4, 1, 9, 0, 0, 5],
+                   [0, 0, 0, 0, 8, 0, 0, 7, 9]]
+    # When the program asks for input we give like this : 
+            5, 3, 0, 0, 7, 0, 0, 0, 0,6, 0, 0, 1, 9, 5, 0, 0, 0,0, 9, 8, 0, 0, 0, 0, 6, 0, 8, 0, 0, 0, 6, 0, 0, 0, 3,4,
```
Comment (vatsalkesarwani12) [inline Python/other/sudoku_backtracking.py:69]: make this a valid input  
i tried to copy and paste it and the code fails

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

File: Python/other/sudoku_backtracking.py (hunk 5 above)
Comment (rgrupesh) [reply Python/other/sudoku_backtracking.py:69]: You need to manually input numbers on prompt to solve sudoku. You cannot give all at once.

Comment (rgrupesh) [reply Python/other/sudoku_backtracking.py:69]: @plazzy99

**Review by vatsalkesarwani12 — APPROVED** [review]

**Comment by rgrupesh** [comment]
@atarax665 @Kajol-Kumari

File: Python/other/sudoku_backtracking.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    1 9 8 |3 4 2 |5 6 7
+    ------------------------
+    8 5 9 |7 6 1 |4 2 3
+    4 2 6 |8 5 3 |7 9 1
+    7 1 3 |9 2 4 |8 5 6
+    ------------------------
+    9 6 1 |5 3 7 |2 8 4
+    2 8 7 |4 1 9 |6 3 5
+    3 4 5 |2 8 6 |1 7 9
+    """
```
Comment (Kajol-Kumari) [inline Python/other/sudoku_backtracking.py]: please leave one extra blank line at the end of the file.

**Review by Kajol-Kumari — APPROVED** [review]
@rgrupesh very minor change required. Rest LGTM :+1:

File: Python/other/sudoku_backtracking.py (hunk 6 above)
Comment (rgrupesh) [reply Python/other/sudoku_backtracking.py]: done

**Review by atarax665 — COMMENTED** [review]
Resolve merge conflict so that I can merge this PR.

**Comment by rgrupesh** [comment]
> Resolve merge conflict so that I can merge this PR.

Done

**Review by atarax665 — APPROVED** [review]
Looks good to me. :)

**Pull request merged**